sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/CosmosRepository.java

```diff
@@ -36,6 +38,19 @@
      */
     void deleteById(ID id, PartitionKey partitionKey);
 
+    /**
+     * Patches an entity by its id and partition key with CosmosPatchItemRequestOptions
+     *
+     * @param entityToPatch entity to patch, must not be null, must contain {@link org.springframework.data.annotation.Id}
+     * and {@link com.azure.spring.data.cosmos.core.mapping.PartitionKey} field
+     * @param patchOperations patch operations, must not be null.
+     * @param options Optional CosmosPatchItemRequestOptions, e.g. options.setFilterPredicate("FROM products p WHERE p.used = false");
+     * @param <S> type of entity
+     * @return the patched entity
+     * @throws IllegalArgumentException in case the given {@code id} is {@literal null}.
+     */
+    <S extends T> S save(S entityToPatch, CosmosPatchOperations patchOperations, CosmosPatchItemRequestOptions options);
```

Comment:
should we also introduce an API without the requestOptions? `save(S entityToPatch, CosmosPatchOperations patchOperations);`

Reply:
No we will not, we want to maintain the least amount of API functions possible. Allowing the requestOptions to be null is acceptable and follows our best practices in Spring.